### PR TITLE
Cosmic Shiv Rework (from the person the item is dedicated to)

### PR DESCRIPTION
**Pull request by YataYT**
Rework from Yatagarasu (patreon donor for Cosmic Shiv), which now shoots a homing star trail to a target, spawning an invisible, non-damaging aura that sticks to the target, continously spawning extremely fast blades for 5 seconds. I didn't add a limit for how many auras can exist at once, but I added limits to how many can be applied at once. Galaxy Star is no longer a thing, but the texture is used for Cosmic Shiv Trail now, so I renamed the file to match. Not multiplayer-tested since I have no friends and my PC is too beefy to know how laggy it might be for weaker systems.

My intention with this weapon is to be a slow firing projectile-focused weapon that can stack up a bit on one target, and be able to keep up with Yharon, hence the accelerating homing and high speed blades. Feel free to nerf the penetration of the blades if they're too op against Thanatos.

This one was a big step-up from my previous projects (which was basically just making very basic weapons or slightly modifying ExampleMod code, and I'm overall extremely happy with how it went. Code might be pretty bad since I only had 2 weeks of modding experience by the time I finished this, so the code might need some cleaning up, and if the draws are too laggy, feel free to nerf the values a bit (although I tweaked with them a lot to get them looking the best)

### CODE REVIEW

File: Projectiles/Melee/Shortswords/CosmicShivProj.cs [hunk 1]
```diff
@@ -50,31 +57,65 @@ public override void SetVisualOffsets()
...
-            target.AddBuff(ModContent.BuffType<GodSlayerInferno>(), 120);
+            // Congrats on trying to PvP with this
+            target.AddBuff(ModContent.BuffType<GodSlayerInferno>(), 216000);
+            target.AddBuff(ModContent.BuffType<VulnerabilityHex>(), 216000);
+            target.AddBuff(ModContent.BuffType<WhisperingDeath>(), 216000);
+            target.AddBuff(ModContent.BuffType<CrushDepth>(), 216000);
+            target.AddBuff(ModContent.BuffType<Nightwither>(), 216000);
+            target.AddBuff(ModContent.BuffType<VulnerabilityHex>(), 216000);
+            target.AddBuff(ModContent.BuffType<GlacialState>(), 600);           // Not that evil
+            target.AddBuff(BuffID.Stoned, 600);                                 // But just in case you're immune to glacial state
```
Comment (Ozzatron) [inline Projectiles/Melee/Shortswords/CosmicShivProj.cs]: Remove this function.

**Review by Ozzatron — CHANGES_REQUESTED** [review]
Remove the PvP meme function and it looks good.

File: Projectiles/Melee/CosmicShivAura.cs [hunk 2]
```diff
@@ -44,7 +44,7 @@ public override void OnSpawn(IEntitySource source)
         public override void AI()
         {
             // Unexist with target
-            if (!target.active || target is null) {
+            if (!target?.active == true || target is null) {
```
Comment (Ozzatron) [inline Projectiles/Melee/CosmicShivAura.cs]: By flipping these cases around you don't need to guard against target being null twice.
```suggestion
            if (target is null || !target.active) {
```

**Review by Ozzatron — DISMISSED** [review]
Looks good except for that if statement, which I've got handled.

**Review by Ozzatron — APPROVED** [review]

**Review by steviegt6 — APPROVED** [review]

**Pull request merged**